### PR TITLE
chore(gateway): Separate gateway config & use TOML in integration-tests

### PR DESCRIPTION
**Pull request by Finistere**
Instead of adding custom methods to imitate the TOML configuration, integration-tests
can now just use it directly.

Moving the gateway config into a separate crate also allows
`runtime-local` to depend on it directly if relevant, looking at you
Redis.

Had to do a bunch of clone with the config, we should ideally separate it into multiple struct with `#[serde(flatten)]` if TOML supports it

### CODE REVIEW

File: engine/crates/engine-config-builder/src/from_sdl_config.rs [hunk 1]
```diff
@@ -0,0 +1,299 @@
+//! Glue crate between parser-sdl & engine-v2-config
```
Comment (Finistere) [inline engine/crates/engine-config-builder/src/from_sdl_config.rs:1]: copy-pasted what was in `lib` before

File: engine/crates/engine-config-builder/src/from_toml_config.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+use engine_v2_config::VersionedConfig;
```
Comment (Finistere) [inline engine/crates/engine-config-builder/src/from_toml_config.rs:1]: moved the logc from `federated-server`

File: engine/crates/integration-tests/Cargo.toml [hunk 3]
```diff
@@ -23,9 +23,10 @@ engine-config-builder = { path = "../engine-config-builder" }
 expect-test = "1.5"
 futures = "0.3.30"
 gateway-core = { workspace = true, features = ["partial-caching"] }
+gateway-config.workspace = true
 grafbase-graphql-introspection.workspace = true
 graphql-composition.workspace = true
-graphql-federated-graph.path = "../federated-graph"
+graphql-federated-graph = { path = "../federated-graph" }
```
Comment (pimeys) [inline engine/crates/integration-tests/Cargo.toml:29]: why not .workspace for this too?

File: gateway/crates/config/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+version = "0.6.1"
+
+[lints]
+workspace = true
+
+[dependencies]
+ascii = { version = "1.1.0", features = ["serde"] }
+duration-str = "0.11.0"
+engine.workspace = true
+grafbase-telemetry = { workspace = true, features = ["tower", "otlp"] }
```
Comment (pimeys) [inline gateway/crates/config/Cargo.toml:17]: I'm not really sure about this. Should we actually just have the configuration options for telemetry in the config crate, so we don't get this kind of crazy dependency chain here...

File: gateway/crates/config/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+repository.workspace = true
+version = "0.6.1"
+
+[lints]
+workspace = true
+
+[dependencies]
+ascii = { version = "1.1.0", features = ["serde"] }
+duration-str = "0.11.0"
+engine.workspace = true
```
Comment (pimeys) [inline gateway/crates/config/Cargo.toml:16]: It feels kind of weird to depend on engine in a config crate that configures engine.

File: gateway/crates/config/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+grafbase-telemetry = { workspace = true, features = ["tower", "otlp"] }
+http.workspace = true
+parser-sdl = { version = "0.1.0", path = "../../../engine/crates/parser-sdl" }
```
Comment (pimeys) [inline gateway/crates/config/Cargo.toml:19]: Should whatever needs this crate to actually get moved to parser-sdl? Would be great if the whole workspace depends on the config, not the other way around.

Comment (pimeys) [reply gateway/crates/config/Cargo.toml:19]: Also workspace would be nicer here than path.

File: gateway/crates/federated-server/src/server.rs [hunk 7]
```diff
@@ -61,24 +59,7 @@ pub async fn serve(
...
-                header_rules: config.headers,
-                rate_limit: config.gateway.rate_limit,
-                timeout: config.gateway.timeout,
-                entity_caching: config.entity_caching,
-            },
-            otel_reload,
-            sender,
-        )
-        .await?;
+    fetch_method.start(&config, otel_reload, sender).await?;
```
Comment (pimeys) [inline gateway/crates/federated-server/src/server.rs:62]: This one here I need to change again, and is a big merge conflict for this PR and mine: the gateway config has the config file location and boolean if hot reloading is enabled on my side. I really also don't want to introduce two new parameters to this function, especially if one of them is boolean.

What I think I'll do is I'll add a new `GatewayConfig` with the config, otel reload, sender, config path and hot reload.

**Review by pimeys — APPROVED** [review]
good stuff.

some paper cuts ahead with merge conflicts.

File: gateway/crates/config/Cargo.toml (hunk 5 above)
Comment (Finistere) [reply gateway/crates/config/Cargo.toml:16]: fixed

File: gateway/crates/config/Cargo.toml (hunk 6 above)
Comment (Finistere) [reply gateway/crates/config/Cargo.toml:19]: did the workspace, but parser-sdl is kind of required anyway today. We're passing through SDL config to build the engine configuration today. And some tests for entity caching directly rely on stuff from `parser-sdl`. So well not fixing it in this PR

**Pull request merged**